src/engine/SCons/Node/__init__.py

```diff
@@ -1131,11 +1132,10 @@ def new_ninfo(self):
         return ninfo
 
     def get_ninfo(self):
-        try:
-            return self.ninfo
-        except AttributeError:
-            self.ninfo = self.new_ninfo()
+        if self.ninfo is not None:
```

Comment:
pretty sure you can get away with "If self.ninfo" ? None equates to false and any false here would be an isssue?

Reply:
This is based on [PEP8's Programming Recommendations](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)

I will change it if you want.

Reply:
Gotcha. I'll defer to PEP8 here.